cborg/src/Codec/CBOR/FlatTerm.hs

```diff
@@ -141,8 +148,118 @@ convFlatTerm (Enc.TkFloat16  f  ts) = TkFloat16   f : convFlatTerm ts
 convFlatTerm (Enc.TkFloat32  f  ts) = TkFloat32   f : convFlatTerm ts
 convFlatTerm (Enc.TkFloat64  f  ts) = TkFloat64   f : convFlatTerm ts
 convFlatTerm (Enc.TkBreak       ts) = TkBreak       : convFlatTerm ts
+convFlatTerm (Enc.TkEncoded  bs ts) = decodePreEncoded bs
+                                                   ++ convFlatTerm ts
 convFlatTerm  Enc.TkEnd             = []
 
+--------------------------------------------------------------------------------
+
+decodePreEncoded :: BS.ByteString -> FlatTerm
+decodePreEncoded bs0 =
+    ST.Lazy.runST (provideInput bs0)
+  where
+    provideInput :: BS.ByteString -> ST.Lazy.ST s FlatTerm
+    provideInput bs
+      | BS.null bs = return []
+      | otherwise  = do
+          next <- ST.Lazy.strictToLazyST $ do
+              Read.Partial k <- Read.deserialiseIncremental decodeTermToken
```

Comment:
This seems a tricky place, @dcoutts could you explain why the partial pattern match is safe?

Reply:
It cannot be `Done` here because of the nature of `decodeTermToken` which always consumes some input. But it could actually be fail, and we probably should provide a better error message, like we do in the case below. I'll improve it to make it explicit. Thanks.

Reply:
Added a comment about why it cannot be Done or Fail. So original code is fine. Comment explains why. Thanks, it was not trivial, deserves an explanation.